charts/digma-ng/templates/kafka/kafka-cr.yaml

```diff
@@ -0,0 +1,153 @@
+  roles:
+    - controller
+  storage:
+    type: persistent-claim
+    size: {{ .Values.kafka.controller.persistence.size | default "10Gi" }}
+    {{- if .Values.kafka.controller.persistence.storageClass }}
+    class: {{ .Values.kafka.controller.persistence.storageClass }}
+    {{- end }}
+    deleteClaim: {{ .Values.kafka.controller.persistence.deleteClaim | default false }}
+  resources:
+    requests:
+      memory: {{ .Values.kafka.controller.resources.requests.memory | default "1Gi" }}
+      cpu: {{ .Values.kafka.controller.resources.requests.cpu | default "250m" }}
+    limits:
+      memory: {{ .Values.kafka.controller.resources.limits.memory | default "2Gi" }}
+      cpu: {{ .Values.kafka.controller.resources.limits.cpu | default "500m" }}
+  {{- if .Values.kafka.controller.jvmOptions }}
+  jvmOptions:
+    {{- toYaml .Values.kafka.controller.jvmOptions | nindent 4 }}
+  {{- end }}
+  template:
+    pod:
+      metadata:
+        labels:
+          {{- with .Values.kafka.controller.podLabels }}
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+        annotations:
+          {{- with .Values.kafka.controller.podAnnotations }}
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+      {{- with .Values.kafka.controller.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.kafka.controller.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.kafka.controller.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+---
+# Main Kafka Resource
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: {{ include "common.names.fullname" . }}-kafka
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: kafka
+  annotations:
+    strimzi.io/kraft: "enabled"
+    strimzi.io/node-pools: "enabled"
+spec:
+  kafka:
+    version: {{ .Values.kafka.version.kafka | default "4.0.0" }}
+    metadataVersion: {{ .Values.kafka.version.metadata | default "4.0-IV2" }}
+    listeners:
+      - name: plain
+        port: {{ .Values.kafka.service.ports.client | default 9092 }}
+        type: internal
+        tls: false
+    config:
+      # Replication settings
+      offsets.topic.replication.factor: {{ .Values.kafka.replication.factor | default 1 }}
+      transaction.state.log.replication.factor: {{ .Values.kafka.replication.factor | default 1 }}
+      default.replication.factor: {{ .Values.kafka.replication.factor | default 1 }}
+      min.insync.replicas: {{ .Values.kafka.replication.minInsyncReplicas | default 1 }}
+      # Log retention settings
+      log.retention.check.interval.ms: 120000
+      log.roll.ms: 120000
+      log.retention.ms: 600000
+      # Topic and performance settings
+      auto.create.topics.enable: true
```

Comment:
lets' set `auto.create.topics.enable` to `false`, so our code is the one that creates the topics with our desired partitions number.